doc/driver.mld

```diff
@@ -1,1124 +1,415 @@
-                  Fpath.basename f |> String.capitalize_ascii = dep_name)
-                all_units
-            with
-            | None -> acc
-            | Some (parent, lib, dep_path, impl) ->
-                let file = best_file dep_path in
-                rec_compile ?impl parent lib file @ acc)
-          [] deps.deps
-      in
-      let ignore_output = parent = "deps" in
-      compile_src file impl ~ignore_output ();
-      compile file ~parent:lib ~ignore_output [];
-      { file = output; ignore_output; source = impl; assets = [] } :: files
-  in
-  source_tree
-  :: List.fold_left
-       (fun acc (parent, lib, dep, impl) ->
-         acc @ rec_compile ?impl parent lib (best_file dep))
-       [] all_units
-  @ mld_odocs
-]}
+{2 The link phase}
 
-Linking is now straightforward. We link all [odoc] files.
-
-{[
-let src_file file =
-  let fdir, fname = Fpath.split_base file in
-  let fname = Fpath.v ("impl-" ^ Fpath.to_string fname) in
-  Fpath.( // ) fdir fname
-
-let link_all odoc_files =
-  List.map
-    (fun ({ file = odoc_file; ignore_output; source; _ } as unit) ->
-      if Option.is_some source then ignore (link ~ignore_output (src_file odoc_file));
-      ignore (link ~ignore_output odoc_file);
-      { unit with file = Fpath.set_ext "odocl" odoc_file })
-    odoc_files
-]}
+The link phase requires the directory of the [compile] phase to generate its set
+of [.odocl] files. This phase resolves references and canonicals.
+
+A generic link command is:
 
-Now we simply run [odoc html-generate] over all the resulting [odocl] files.
-This will generate sources, as well as documentation for non-hidden units.
-We notify the generator that the JavaScript file to use for search is [index.js].
-
-{[
-let generate_all odocl_files =
-  let search_uris = [ Fpath.v "minisearch.js"; Fpath.v "index.js" ] in
-  List.iter
-    (fun { file; ignore_output = _; source; assets } ->
-      ignore (html_generate ~assets ~search_uris file None);
-      match source with
-      | None -> ()
-      | Some source ->
-          ignore (html_generate (src_file file) (Some (relativize source))))
-    odocl_files;
-  support_files ()
+{@shell[
+  $ odoc link
+      -I <dir1> -I <dir2>
+      -P <pname1>:<pdir1> -P <pname2>:<pdir2>
+      -L <lname1>:<ldir1> -L <lname2>:<ldir2>
+      <path/to/file.odoc>
 ]}
 
+- [<path/to/file.odoc] is the input [.odoc] file. The result of this command is
+  [path/to/file.odocl]. This path was determined by [--output-dir] and
+  [--parent-id] from the link phase, and it is important for the indexing phase
+  that it stays in the same location.
+
+- [-P <name>:<dir>] are used to list the "named page-units", used to resolve
+  references such as [{!/ocamlfind/index}].
+
+- [-L <name>:<dir>] are used to list the "named module-units", used to resolve
+  references such as [{!/findlib.dynload/Fl_dynload}].
+
+- [-I <dir>] are used to resolve non-references, and should include the same set
```

Comment:
I'm not sure what a non-reference is? Especially if it's still something that can be resolved?

Reply:
This is indeed confusing. I'm not even sure what I wanted to say...